frontend/apps/reader/modules/readerannotation.lua

```diff
@@ -79,17 +79,17 @@ function ReaderAnnotation:getAnnotationsFromBookmarksHighlights(bookmarks, highl
     for i = #bookmarks, 1, -1 do
         table.insert(annotations, self:buildAnnotation(bookmarks[i], highlights, init))
     end
-    if #annotations > 1 then
-        local sort_func = self.ui.rolling and function(a, b) return self:isItemInPositionOrderRolling(a, b) end
-                                           or function(a, b) return self:isItemInPositionOrderPaging(a, b) end
-        table.sort(annotations, sort_func)
+    if init then
+        self:sortItems(annotations)
     end
     return annotations
 end
 
 function ReaderAnnotation:onReadSettings(config)
     local annotations = config:readSetting("annotations")
     if annotations then
+        local needs_update = config:isTrue("annotations_needs_update")
```

Comment:
May be mention why this setting:
`-- KOHighlight may set this key when it has merged annotations from different sources: we want to make sure they are updated and sorted`